types/magicmirror-module/index.d.ts

```diff
@@ -0,0 +1,83 @@
+// Type definitions for non-npm package magicmirror-module 2.16
+// Project: https://magicmirror.builders/
+// Definitions by: Jalibu <https://github.com/jalibu>
+//                 MichMich <https://github.com/MichMich>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace Module {
+    /* tslint:disable:no-unnecessary-generics */
+    function register<T>(
+        moduleName: string,
+        moduleProperties: ThisType<NonNullable<ModuleProperties<T>>> &
+            Partial<ModuleProperties<T>>,
+    ): void;
+
+    class ModuleProperties<T> {
+        readonly name: string;
+        readonly identifier: string;
+        readonly hidden: boolean;
+        readonly data: { classes: string; file: string; path: string; header: string; position: string };
+        readonly lockStrings: string[];
+        config: T;
+        defaults: T;
+        requiresVersion: string;
+
+        // Subclassable methods
+        init: () => void;
+        loaded: (callback?: () => void) => void;
+        start: () => void;
+        getScripts: () => string[];
+        getStyles: () => string[];
+        getTranslations: () => { [key: string]: string };
+        getTemplate: () => string;
+        getTemplateData: () => any;
+        getDom: () => HTMLElement;
+        getHeader: () => string;
+        notificationReceived: (notification: string, payload: any, sender: ModuleProperties<any>) => void;
+        socketNotificationReceived: (notification: string, payload: any) => void;
+        suspend: () => void;
+        resume: () => void;
+
+        // Instance methods
+        readonly file: (filename: string) => string;
+        readonly updateDom: (speed: number) => void;
+        readonly sendNotification: (notification: string, payload: any) => void;
+        readonly sendSocketNotification: (notification: string, payload: any) => void;
+        readonly hide: NonNullable<(speed?: number, callback?: () => void, options?: { lockString: string }) => void>;
+        readonly show: (
+            speed?: number,
+            callback?: () => void,
+            options?: { lockString?: string; force?: boolean; onError?: () => void },
+        ) => void;
+        readonly translate: (identifier: string, variables?: any) => string;
+
+        [key: string]: any;
+    }
+}
+
+declare namespace node_helper {
+    function create(object: NodeHelperModule): void;
+
+    type NodeHelperModule = {
+        init?: () => void;
+        start?: () => void;
+        stop?: () => void;
+        socketNotificationReceived?: (notification: string, payload: any) => void;
+        sendSocketNotification?: (notification: string, payload: any) => void;
+        [key: string]: any;
+    } & ThisType<NodeHelperModule>;
+}
+
+declare const config: {
```

Comment:
this suggests there is a 'config' symbol somewhere when this is loaded, 
is that a case? If not, should this be a type or interface (a contract)

Reply:
Yes, config is a symbol which is available within the runtime scope, provided by the platform